### ISSUE TITLE
null/undefined exception with nested id and parentId properties

### ISSUE THREAD

**Issue by IMilja** [philipstanislaus/performant-array-to-tree#38]
Hi there,
I've tried to use this library for constructing a tree on a application I'm developing and while it has the awesome feature of using nested id and parentId properties I've noticed a potential bug on it.

The data that I'm getting from an API has the following structure:

```
var performantArrayToTree = require("performant-array-to-tree");

const data = [
  {
    id: 1,
    title: "Some title",
    parent: null
  },
  {
    id: 125,
    title: "Some title 2",
    parent: {
      id: 1,
    },
  },
];

console.log(performantArrayToTree.arrayToTree(data, {
    parentId: 'parent.id',
}));
```

Expected result:
```
[
  {
    data: { id: 1, title: "Some title", parent: null },
    children: [
      {
        data: { id: 125, title: "Some title 2", parent: null },
        children: [],
      }
    ]
  }
]
```
Result I've gotten:
`TypeError: Cannot read property 'parentId' of null`

My guess is that the problem is in the following function:
```
function getNestedProperty(item: Item, nestedProperty: string) {
  return nestedProperty.split(".").reduce((o, i) => o[i], item);
}
```

Since it's accessing the object `o` which in my case is `null` with the key of `i` it will throw an exception. A potential fix is to check if `o` is `null` or `undefined` and if it is the reduce function should return `null` as well.
```
function getNestedProperty(item: Item, nestedProperty: string) {
  return nestedProperty.split(".").reduce((o, i) => (o === null || o === undefined) ? null : o[i], item);
}
```

**Comment by philipstanislaus** [comment]
Sorry for the late reply, and thanks for reporting this. 

You are right about the source of this issue, here's the fix: 7540bb2d0c1d7eff850ec5241814f6268ca76471

**Comment by philipstanislaus** [comment]
Published in v1.9.2: https://www.npmjs.com/package/performant-array-to-tree/v/1.9.2
